core/src/main/clojure/xtdb/util.clj

```diff
@@ -627,18 +627,40 @@
 (defn ->kebab-case-kw [s]
   (-> s name str/lower-case (str/replace "_" "-") keyword))
 
-(defn ns-kw->kw [kw]
-  (if-let [ns (namespace kw)]
-    (keyword (str ns "__" (name kw)))
-    kw))
+(defn- ->normal-form [s]
+  (-> s (str/replace "." "$") (str/replace "-" "_")))
 
-(defn kw->ns-kw [kw]
-  (apply keyword (str/split (name kw) #"__")))
+(defn- <-normal-form [s]
+  (-> s (str/replace "$" ".") (str/replace "_" "-")))
 
-(defn ns-symbol->symbol [s]
+(defn kw->normal-form-kw [kw]
```

Comment:
might be a more intent-revealing way to do this with `str/split`, `butlast`/`last`, and `str/join`?